app/src/main/java/com/halim/adam/cashmaster/DatabaseHelper.java

```diff
@@ -61,39 +62,34 @@ public void InsertCategory(String name){
         stmt.executeInsert();
         db.close();
     }
-
     public void InsertIncome(String name, float price, Date date){
         SQLiteDatabase db = this.getWritableDatabase();
-        String sql = "INSERT INTO income (name, price, date) VALUES ('" + name + "', '" + price + "', '" + date +  "');";
+        String sql = "INSERT INTO income (name, price, date) VALUES ('" + name + "', '" + price + "', '" + new SimpleDateFormat("yyyy-MM-dd").format(date) +  "');";
```

Comment:
Some kind of validation should be added to the name variable before inserting it into the database as people can insert sql commands to manipulate the database